python/cog/server/clients.py

```diff
@@ -188,17 +199,24 @@ async def upload_file(self, fh: io.IOBase, url: Optional[str]) -> str:
         # ensure trailing slash
         url_with_trailing_slash = url if url.endswith("/") else url + "/"
 
-        async def chunk_file_reader() -> AsyncIterator[bytes]:
-            while 1:
-                chunk = fh.read(1024 * 1024)
-                if isinstance(chunk, str):
-                    chunk = chunk.encode("utf-8")
-                if not chunk:
-                    log.info("finished reading file")
-                    break
-                yield chunk
+        class ChunkFileReader:
+            async def __aiter__(self) -> AsyncIterator[bytes]:
+                fh.seek(0)
+                while 1:
+                    chunk = fh.read(1024 * 1024)
+                    if isinstance(chunk, str):
+                        chunk = chunk.encode("utf-8")
+                    if not chunk:
+                        log.info("finished reading file")
+                        break
+                    yield chunk
```

Comment:
The previous implementation didn't reset the file handle on subsequent calls which resulted in an `httpx.StreamConsumed` exception. I added a test and implemented this fix which will ensure a new iterator is created for each request.

Reply:
FWIW I believe the function-based implementation would work with a fh.seek(0) at the top of the function, but this is great/cleaner